### PR TITLE
fix(settings): Update reset password warning message

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

* We want to clarify some of the reset password warning wording
* We want to provide a link back to account_recovery_confirm_key on the complete_reset_password page when the user ininitially chooses not to use their key
* We want to show a warning message if we can't determine if the user has a recovery key (instead of an error)
* Also show the warning to users signing in to sync, even if they don't have actively synced devices (signing out from sync removes the device from estimated device count, but does not mean the user does not have synced data)

## This pull request

* Update warning message and associated l10n, stories, tests
* Update CompleteResetPassword page to add link conditionally
* Some styling tweaks

## Issue that this pull request solves

Closes: #FXA-10483

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

For users who have a recovery key but didn't use it OR if we don't know if they have a key:
image

For sync users who do not have a key:
image

## Other information (Optional)

Also fixed an unrelated functional test that was flaky - failed in headless mode but passes in headed.

### CODE REVIEW

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-settings/src/components/ResetPasswordWarning/index.tsx [hunk 1]
```diff
@@ -49,50 +58,80 @@ const ResetPasswordWarning = () => {
...
+              <FtlMsg id="password-reset-warning-use-key-link">
+                <Link
+                  to={`/account_recovery_confirm_key${searchParams}`}
+                  state={locationState}
+                  className="link-blue"
+                  onClick={() =>
+                    GleanMetrics.passwordReset.createNewRecoveryKeyMessageClick()
+                  }
+                >
+                  Use it now to reset your password and keep your data
```
Comment (LZoog) [inline packages/fxa-settings/src/components/ResetPasswordWarning/index.tsx:84]: Tiny nit, this link expands the full width.

File: packages/fxa-settings/src/components/ResetPasswordWarning/index.tsx [hunk 2]
```diff
@@ -49,50 +58,80 @@ const ResetPasswordWarning = () => {
...
+            />
+            <div className="flex flex-col flex-1 gap-1">
+              <FtlMsg id="password-reset-warning-have-key">
+                <p className="font-semibold text-sm">
+                  Have an account recovery key?
+                </p>
+              </FtlMsg>
+              <FtlMsg id="password-reset-warning-use-key-link">
+                <Link
+                  to={`/account_recovery_confirm_key${searchParams}`}
```
Comment (LZoog) [inline packages/fxa-settings/src/components/ResetPasswordWarning/index.tsx:77]: Do we need `searchParams || ''` here if this is undefined, or `?{searchParams}`? (I didn't test, just have seen it be wonky myself, maybe `Link` handles this better than `navigate`)

File: packages/fxa-settings/src/components/ResetPasswordWarning/indes.stories.tsx [hunk 3]
```diff
@@ -1,21 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (LZoog) [inline packages/fxa-settings/src/components/ResetPasswordWarning/indes.stories.tsx:1]: whoops @ this file name 😆

File: packages/fxa-settings/src/components/ResetPasswordWarning/index.tsx [hunk 4]
```diff
@@ -49,50 +58,80 @@ const ResetPasswordWarning = () => {
...
+        <div className="flex items-start gap-4">
+          <IconSyncDevice role="img" className="mt-[2px]" aria-hidden={true} />
+          <div className="flex flex-col flex-1 gap-1">
+            <FtlMsg id="password-reset-previously-signed-in-device-2">
+              <p className="font-semibold text-sm">
+                Have any device where you previously signed in?
+              </p>
+            </FtlMsg>
+            <FtlMsg id="password-reset-data-may-be-saved-locally-2">
+              <p className="text-grey-500">
```
Comment (LZoog) [inline packages/fxa-settings/src/components/ResetPasswordWarning/index.tsx:99]: I double checked that this grey on the yellow passes AA and it does indeed for "normal text", not sure if there's a better tool to check text _this_ small though.

**Review by LZoog — APPROVED** [review]
LGTM!

**Pull request merged**